Index: js-1.8.5/js/src/jsdbgapi.cpp
===================================================================
--- js-1.8.5.orig/js/src/jsdbgapi.cpp
+++ js-1.8.5/js/src/jsdbgapi.cpp
@@ -127,9 +127,11 @@ CompartmentHasLiveScripts(JSCompartment
     JSContext *iter = NULL;
     JSContext *icx;
     while ((icx = JS_ContextIterator(comp->rt, &iter))) {
-#ifdef JS_THREADSAFE
+#ifdef JS_METHODJIT
+# ifdef JS_THREADSAFE
         if (JS_GetContextThread(icx) != currentThreadId)
             continue;
+# endif
 #endif
         for (AllFramesIter i(icx); !i.done(); ++i) {
             JSScript *script = i.fp()->maybeScript();
